File: KuantumCSharp/KuantumOdev/KaranlikMadde.cs
using System;

// tehlikeli nesne (IKritik uygulanır)
public class KaranlikMadde : KuantumNesnesi, IKritik
{
    public KaranlikMadde(string id)
    {
        ID = id;
        Stabilite = 70.0; // örnek başlangıç stabilitesi
        TehlikeSeviyesi = new Random().Next(1, 11);
    }

    public override void AnalizEt()
    {
        Stabilite -= 15; // stabilite 15 birim düşer
    }

    public void AcilDurumSogutmasi()
    {
        Stabilite += 50;
        Console.WriteLine("Soğutma yapıldı, stabilite yenilendi.");
    }
}

File: KuantumJava/KaranlikMadde.java
// tehlikeli nesne (IKritik uygulanır)
public class KaranlikMadde extends KuantumNesnesi implements IKritik {
    public KaranlikMadde(String id) {
        super(id);
    }
    @Override
    public void analizEt() throws KuantumCokusuException {
        setStabilite(getStabilite() - 15);     // stabilite 15 azalır
    }
    @Override
    public void acilDurumSogutmasi() {
        try {
            setStabilite(getStabilite() + 50);
            System.out.println("Soğutma yapıldı, stabilite yenilendi.");
        } catch (KuantumCokusuException e) {
            // 100'den fazla olmaz, hata çıkmaz
        }
    }
}
